nativelib/src/main/resources/scala-native/delimcc.c

```diff
@@ -15,11 +15,11 @@
 #define ASM_JMPBUF_SIZE 192
 #define JMPBUF_STACK_POINTER_OFFSET (104 / 8)
 #elif defined(__x86_64__) &&                                                   \
-    (defined(__linux__) || defined(__APPLE__)) // x86-64 linux and macOS
+    (defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__)) // x86-64 linux, macOS and FreeBSD
 #define ASM_JMPBUF_SIZE 72
 #define JMPBUF_STACK_POINTER_OFFSET (16 / 8)
 #elif defined(__i386__) &&                                                     \
-    (defined(__linux__) || defined(__APPLE__)) // x86 linux and macOS
+    (defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__)) // x86 linux, macOS and FreeBSD
```

Comment:
Has this change been tested on 32-bit FreeBSD?

By my understanding, 32 bit FreeBSD is fully supported
up to and including FreeBSD 12. In FreeBSD 13 and above, it became " Tier 2: Developmental and Niche Architectures".

If you are using & testing 32 bit FreeBSD, especially on FBSD > 12,  the change is fine.  If not, my suggestion is 
a separate branch which errors out on 32-bit FreeBSD (i.e. unsupported).   

That would allow testing and maintenance to focus on 64-bit, where such is struggling already.

Reply:
Note that any changes that don't harm the mainline code towards the goal of first rate FreeBSD support should be welcome but I don't make that call. Also consider adding Setup documentation for FreeBSD and not supporting Boehm initially could be avoided by a simple "Boehm is not supported". Having a way via Docker or something to add to CI would be a great gain too even if some tests are "assumed" away. Even having one configuration that works would go a long way.

Reply:
Mostly agreed. Having an at least weekly FreeBSD CI run has been a goal (perhaps for a very small set of
people) since FreeBSD quasi-support was merged into mainline. I say "quasi" because it had no CI tests.

The problem has been that no one has had the time to make the required changes to CI.  

Given all the work that Wojciech has has put into rational CI runs over the past months/years,
the job should be easier, but such work still takes devo cycles.